src/unitxt/inference.py

```diff
@@ -449,7 +457,10 @@ def _initialize_wml_client(self):
             self.credentials = self._read_wml_credentials_from_env()
 
         client = APIClient(credentials=self.credentials)
-        client.set.default_project(self.credentials["project_id"])
+        if "space_id" in self.credentials:
```

Comment:
I see that there may be an issue when a user passes credentials directly to the engine as an instance of `ibm_watsonx_ai.credentials.Credentials` (which is allowed). This class does not have either `space_id` or `project_id` as its attributes. 
So one way could be to require `credentials` to be always a dictionary. The other solution could be to sill allows passing an instance of `Credentials` but also demand providing any of `project_id` or `space_id` directly to the engine

As for inference with logprobs, it looks good to me

Reply:
@pawelknes - The project_id/space_id in the credentials are set by the PROJECT_ID/SPACE_ID env variables, if they exist. We can add a check to set them according to WML_PROJECT_ID/WML_SPACE_ID env variables if the credentials object doesn't contain either project_id or space_id.

Reply:
Alright, there is also another PR with minor fixes to the engine so that'll resolve any related issue anyway